packages/storage/src/indexed-db/index.ts

```diff
@@ -802,4 +784,12 @@ export class IndexedDb implements IndexedDbInterface {
       output: Value.fromJson(result.output),
     };
   }
+
+  async hasNativeAssetBalance(): Promise<boolean> {
+    const key = 'KeqcLzNx9qSH5+lcJHBB9KNW+YPrBk5dKzvPMiypahA=';
+    const jsonArray = await this.db.getAllFromIndex('SPENDABLE_NOTES', 'assetId', key);
+    if (jsonArray.length === 0) return false;
+
+    return true;
```

Comment:
you may have `notes` but they have all been spent, you should probably only get the unspent `notes` here

Reply:
I switched to using `viewService.balances(new BalancesRequest({ accountFilter: addressIndex, assetIdFilter: stakingAssetId }), ctx)` to retrieve the assetId balance.

Reply:
@Valentine1898 further testing revealed that this using `viewService.balances` was in fact much slower because it's iterating over all the spendable notes in indexdb, rather than just the subset of UM notes we care about.

Reply:
I've noticed this call takes ~`500ms`

Reply:
on closer inspection, making this call incurs ~500ms - 1s in additional overhead that noticeably delayed the transaction approval popup. The `balances` query has extraneous logic and iterates over all the spendable notes rather than just inspecting the relevant staking asset notes. I instead opted for a _custom_ note processing scheme which takes a negligible amount of time to execute.